src/server/utilities/canAdvertise/index.js

```diff
@@ -0,0 +1,8 @@
+const canAdvertise = headers => {
+  const isAdvertiseCombine = headers['x-ip_is_advertise_combined'];
+  const isUK = headers['x-bbc-edge-isuk'];
+
+  return isAdvertiseCombine === 'yes' || isUK === 'no';
```

Comment:
This is cleaner, thanks! Although I'm not sure this is _quite_ the logic we want, we only want to check the `'x-bbc-edge-isuk'` if `'x-ip_is_advertise_combined'` isn't present _at all_ - if `'x-ip_is_advertise_combined'` is `no`, this code will then check the `'x-bbc-edge-isuk'` header as well. If `'x-ip_is_advertise_combined'` tells us that we shouldn't show an ad, then we trust that more than `'x-bbc-edge-isuk'`, which we are using here as a backup. Sorry, it's a bit confusing but is the state of things at the moment!

Reply:
Oh, okay, I'll update this logic then. Just to confirm, will this be the correct logic?
```
  if (isAdvertiseCombine === 'yes' || (!isAdvertiseCombine && isUK === 'no')) {
    return true;
  }

  return false;
```

Reply:
I believe that what we want to do is something like this, where we just check for the `'x-bbc-edge-isuk'` header if `x-ip_is_advertise_combined` isn't present, right?
```
 const isAdvertiseCombine = headers['x-ip_is_advertise_combined']; 

if(isAdvertiseCombine && isAdvertiseCombine === 'yes'){
       return true;
} else {
     const isUK = headers['x-bbc-edge-isuk']; 
     if(isUK && isUK === 'no'){
        return true;
     }
}

return false;

``

Reply:
I think something along these lines, with a comment explaining why we are checking both values, but whatever implementation seems easiest to follow for you!

```
const isAdvertiseCombined = headers['x-ip_is_advertise_combined'];
const isUK = headers['x-bbc-edge-isuk'];

const canAdvertise = typeof isAdvertiseCombine !== 'undefined' ? isAdvertiseCombine === 'yes' :  isUK === 'no'
```

Reply:
Denis' approach is verbose  but is easier to understand and wouldn't need any comments to explain. So I'd use that